lib/include/openamp/virtio.h

```diff
@@ -499,6 +511,51 @@ static inline int virtio_reset_device(struct virtio_device *vdev)
 	return 0;
 }
 
+/**
+ * @brief Allocate buffer from the virtio device
+ *
+ * @param vdev	Pointer to virtio device structure.
+ * @param buf	Pointer to the allocated buffer.
+ * @param size	Allocated buffer size.
+ * @param align	Allocated buffer alignment.
+ *
+ * @return 0 on success, otherwise error code.
+ */
+static inline int virtio_alloc_buf(struct virtio_device *vdev, void **buf,
+				   size_t size, size_t align)
+{
+	if (!vdev || !buf)
+		return -EINVAL;
+
+	if (!vdev->mmops || !vdev->mmops->alloc)
+		return -ENXIO;
+
+	*buf = vdev->mmops->alloc(vdev, size, align);
+	if (!*buf)
+		return -ENOMEM;
+	return 0;
+}
+
+/**
+ * @brief Free buffer to the virtio device
```

Comment:
Perhaps it is obvious for us, but as there libmetal also provides metal_allocate_memoryI would add the information
 `Free the buffer allocated by \ref virtio_alloc_buf from the virtio device`

Reply:
Done